.github/workflows/docker-build.yaml

```
@@ -0,0 +1,18 @@
+name: docker-build
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout repo
+      uses: actions/checkout@main
```

```suggestion
      uses: actions/checkout@v5
```
https://github.com/actions/checkout/releases/tag/v5.0.0